Issue 2694 update schema design docs (#3215)
IBM/FHIR: fhir-persistence-schema/docs/SchemaToolUsageGuide.md

# IBM FHIR Server - Schema Deployment

This document details how to deploy the IBM FHIR Server schema and apply release upgrades. This document also describes how the schema changes are managed.

The schema tool generates the following object types that require management: 

- DB2 PACKAGE
- TABLESPACES
- SCHEMA 
- GLOBAL VARIABLE 
- INDEX 
- PERMISSION 
- PROCEDURE 
- SEQUENCE 
- TABLE
- TABLE CONSTRAINT 

For details on the schema design, refer to the [Schema Design](https://github.com/IBM/FHIR/tree/master/fhir-persistence-schema/docs/SchemaDesign.md) document.

----------------------------------------------------------------
## Database Support

| Database   | Version   | Support |
|------------|-----------|-----------------------------------|

> | DB2        |      11.5 | Suitable for production. Supports multi-tenancy. |

Reviewer: having "suitable for production" here and not in PostgreSQL still seems to imply this one is somehow more mature. 
```suggestion
| DB2        |      11.5 | Supports multiple tenants in the same db schema. |
```

Author: fixed

> | PostgreSQL |        12 | Single tenant per database. |

Reviewer: I think the versions listed here are our 'minimum' versions, right?
For example, we can (and do) run with PostgreSQL 13 and 14.  Maybe we can make it more clear?

Author: done

### Running the CLI

There are three core projects which are referenced: 

| Project | Description |
|----------|---------------------------------------------|
| `fhir-model` | The HL7 FHIR model - generated Java code |
| `fhir-database-utils` | The SQL/database constructs used to create SQL compliant statements |
| `fhir-persistence-schema` | Uses database-utils to define a physical data model for storing FHIR resources and supporting the FHIR API |

To deploy and manage the schema on a target database, the `fhir-persistence-schema` project includes a Main class that can apply and manage the schema updates.

``` shell

> java -jar ./fhir-persistence-schema-${VERSION}-cli.jar [OPTIONS]

Reviewer 2: I like the tldr approach

> java -jar ./fhir-persistence-schema-${VERSION}-cli.jar [OPTIONS]

Author: yep, good idea

> java -jar ./fhir-persistence-schema-${VERSION}-cli.jar [OPTIONS]

Author: I created a TL;DR version for PostgreSQL in the README.

|--force-unused-table-removal||Forces the removal of unused tables - DomainResource, Resource|
| --force || Do not skip schema update process when the whole-schema-version matches.|
|--prop|name=value|name=value that is passed in on the commandline|
|--pool-size|poolSize|poolsize used with the database actions|
|--drop-schema-oauth||drop the db schema used by liberty's oauth/openid connect features|
|--drop-schema-batch||drop the db schema used by liberty's java-batch feature"|
|--drop-schema-fhir||drop the schema set by '--schema-name'|
|--update-schema||deploy or update the schema set by '--schema-name', *deprecated* use --update-schema-fhir|
|--update-schema-fhir|schemaName|Updates the FHIR Data Schema|
|--update-schema-batch|schemaName|Updates the Batch Schema|
|--update-schema-oauth|schemaName|Updates the OAuth Schema|
|--create-schemas||create the database schemas for batch, oauth, and the fhir schema set by '--schema-name'|
|--create-schema-fhir|schemaName|Create the FHIR Data Schema|
|--create-schema-batch|schemaName|Create the Batch Schema|
|--create-schema-oauth|schemaName|Create the OAuth Schema|

> FHIR® is the registered trademark of HL7 and is used with the permission of HL7.

Reviewer 2: this move has consequences in other guides. Please be sure to update their links.  There are also other links outside of our control that no longer function.

Author: fixed
